### PR TITLE
Set the message stream as a header so we pass it to the API properly

### PR DESCRIPTION
**Pull request by nickhammond**
For #90.

### CODE REVIEW

File: spec/unit/postmark/mail_message_converter_spec.rb [hunk 1]
```diff
@@ -377,13 +377,17 @@
...
     context 'when not set' do
-      specify { expect(mail_message.to_postmark_hash.keys).not_to include('MessageStream') }
+      specify { expect(subject.new(mail_message).run).not_to include('MessageStream') }
     end
 
     context 'when set' do
-      before { mail_message.message_stream = 'weekly-newsletter' }
+      before do
+        mail_message.message_stream = 'weekly-newsletter'
+    end
```
Comment (tomazy) [inline spec/unit/postmark/mail_message_converter_spec.rb:386]:
```suggestion
      end
```

File: spec/unit/postmark/mail_message_converter_spec.rb [hunk 2]
```diff
@@ -377,13 +377,17 @@
...
 
-      specify { expect(mail_message.to_postmark_hash).to include('MessageStream' => 'weekly-newsletter') }
+      it 'adds message stream as a mail header' do
```
Comment (tomazy) [inline spec/unit/postmark/mail_message_converter_spec.rb:388]:
```suggestion
      it 'passes message stream to the API call' do
```
or something like that. This is not a "mail header"

**Comment by nickhammond** [comment]
@tomazy Let me know if anything else comes to mind, thinking the metadata change will be in another PR. I've dropped CI coverage for 1.8.7 which we already did with `postmark-rails`.

**Comment by tomazy** [comment]
I'm a bit worried about the impact of dropping support for 1.8.7 in a minor release. I don't know if there's any but I'd rather avoid it if possible. Or should it be a major release in such case?

**Comment by nickhammond** [comment]
@tomazy True, it'd be nice to keep it in there. I just noticed the [failed build](https://travis-ci.org/github/wildbit/postmark-gem/builds/725363707) against 1.8.7 only for something unrelated, seeing if we can just fix it for 1.8.7.

**Comment by nickhammond** [comment]
Commits squashed and 1.8.7 is passing again.

**Review by tomazy — APPROVED** [review]
lgtm! Thank you!

**Pull request merged**